README.md

```diff
@@ -7,27 +7,179 @@
 
 # scrapbook
 
-A library for recording a notebook's data values and reading these recorded data
-values at a later time.
+A library for recording a notebook's data values and reading these recorded data values at a later time.
 
-**scrapbook** provides a robust and flexible recording schema. This library is
-planned to replace [papermill](https://papermill.readthedocs.io)'s existing
-`record` functionality.
+Namely scrapbook lets you:
+- **persist** data in a notebook
+- **highlight** named displays in notebooks
+- **recall** any persisted scrap of data or highlight
+- **summarize collections** of notebooks
 
 ## Use case
 
 Notebook users may wish to record data produced during a notebook execution.
 This recorded data can then be read to be used at a later time or be passed to
 another notebook as input.
 
-## papermill's record feature
+## API Calls
+
+Scrapbook adds a few basic api commands which enable saving and retrieving data.
+
+### glue
+
+Records a value (scrap) in the given notebook cell.
+
+The recorded value can be retrieved during later inspection of the output notebook.
 
-### Background
+```python
+sb.glue("hello", "world")
+sb.glue("number", 123)
+sb.glue("some_list", [1, 3, 5])
+sb.glue("some_dict", {"a": 1, "b": 2})
+sb.glue("non_json", df, 'arrow')
+```
+
+The scrapbook library can be used later to recover recorded values by reading the output notebook
+
+```python
+nb = sb.read_notebook('notebook.ipynb')
+nb.scraps
+```
+
+The storage format of the scraps is implied by the value type any registered data translators, but can be overwritten by setting the `storage` argument to a particular translator's registered name (e.g. `"json"`).
+
+This data is persisted by generating a display output with a special media type identifying the content storage format and data. These outputs are not visible in notebook rendering but still exist in the document. Scrapbook then can rehydrate the data associated with the notebook in the future by reading these cell outputs. 
+
+### highlight
+
+Display an object with the reference `name` in a retrievable manner. Unlike `glue` this is intended to generate a visible display output for notebook interfaces to render.
+
+```python
+sb.highlight("hello", "Hello World")
+sb.highlight("sharable_png", IPython.display.Image(filename=get_fixture_path("sharable.png")))
+```
+
+Like scraps these can be retrieved at a later time, though they don't cary any actual data, just the display result of some object.
```

Comment:
Just from reading this sentence I am not sure what the use for a highlight is when retrieving them. Maybe a short "retrieve a highlight from a previous notebook and display it (for example to show a summary plot from another notebook)" which is I think the use case for highlights?

Reply:
Highlight is feeling off the mark to me. Scribe, sketch a snap perhaps.

Reply:
I do like `snaps` better than `highlights`. Is it too close to `scraps` to cause confusion? I like it, but my gut is telling me that someone new to the abstraction would trip over the names. Though, since we have `glue` vs `snap` that's a pretty good distinction on the caller side. Good with giving it a try though if your two think it would be easily distinctive.

Should we lean on the underlying primitive name? `display`? We're essentially just labeling / collecting ipython displays.

Another option is to only have `glue`  and have a storage type called `display` which will render the object. This makes the caller API simpler and groups all the elements together as tracking `scraps`. There could be functions on the Notebook `displayed_scraps` and `hidden_scraps` to collect those scraps which did or didn't render in the notebook. I've been thinking this is more amendable to KISS pattern and easier to explain to users. Thoughts?

Reply:
Given combining the two will be a bit expensive in time and we can change it for 0.2.0 -- I'm going to do the change to sketch/snap paradigm in prep for 0.1.0